web-api/workflow-terraform/migration/main/lambdas/migrations/0001-add-associated-judge-id.ts

```diff
@@ -18,62 +19,33 @@ const isRecordToUpdate = item => {
   return isCaseRecord(item) || isCaseDeadline(item) || isWorkItem(item);
 };
 
-async function getAllJudgeRecords(documentClient) {
-  const scanParams = {
-    ExpressionAttributeNames: {
-      '#n': 'name',
-      '#r': 'role',
-    },
-    ExpressionAttributeValues: {
-      ':prefix': 'user|',
-      ':role1': 'judge',
-      ':role2': 'legacyJudge',
-    },
-    FilterExpression:
-      'begins_with(pk, :prefix) AND begins_with(sk, :prefix) AND (#r = :role1 OR #r = :role2)',
-    ProjectionExpression: 'pk, sk, #r, #n, userId',
-    TableName: process.env.SOURCE_TABLE!,
-  };
-
-  const items: { userId: string; name: string }[] = [];
-  let lastEvaluatedKey = null;
-
-  do {
-    const params = lastEvaluatedKey
-      ? { ...scanParams, ExclusiveStartKey: lastEvaluatedKey }
-      : scanParams;
-
-    const result = (await documentClient.scan(params).promise()) as unknown as {
-      Items: { userId: string; name: string }[];
-      LastEvaluatedKey?: any;
-    };
-
-    items.push(...result.Items);
-    lastEvaluatedKey = result.LastEvaluatedKey;
-  } while (lastEvaluatedKey);
-
-  return items;
-}
-
 export const migrateItems = async (
   items: any[],
-  documentClient: AWS.DynamoDB.DocumentClient,
+  _,
+  applicationContext: ServerApplicationContext,
 ) => {
-  const judgeRecords = await getAllJudgeRecords(documentClient);
-
-  const judgesMap = judgeRecords.reduce((accumulator, judge) => {
-    accumulator[judge.name] = judge.userId;
-    return accumulator;
-  }, {});
-
   const itemsAfter: TDynamoRecord[] = [];
+  let judgesMap: { [key: string]: string } | null = null;
```

Comment:
with `judgesMap` scoped to just this function, it will be looked up every time this function is called, and this function could be called many times per segment as the lambda paginates through records. its value should not change throughout the course of this lambda running. does it make sense to lift it out of this function so it is only looked up once?